common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java

```diff
@@ -112,7 +116,11 @@ public void setSecretKey(byte[] rawKey) {
         if (rawKey == null || rawKey.length != SECRET_RAW_KEY_LENGTH) {
             throw new IllegalArgumentException("rawKey");
         }
-
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            Logger.warn(":setSecretKey", "You're using setSecretKey in a version of android " +
+                    "that supports keyStore functionality.  Consider not doing this, as it only exists " +
+                    "for devices with an SDK lower than " + Build.VERSION_CODES.JELLY_BEAN_MR2);
```

Comment:
and are we doing this right now?

Reply:
We support generating our own key, it happens here:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/9a09e2e4a1fe714773a2b679d054b311d2c56f86/common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java#L666

From a customer investigation we were working on earlier, we found an app that was calling this method at startup: we suspect it's causing issues with SSO loss due to the host-app [potentially] providing mismatched keys

Reply:
Cool.....actually how does user defined key currently work for MSAL? It is only for ADAL right?

Reply:
MSAL reads the value from the AuthenticationSettings.INSTANCE:

        // Try to get user defined key (ADAL/MSAL).
        if (AuthenticationSettings.INSTANCE.getSecretKeyData() != null) {
            setBlobVersion(VERSION_USER_DEFINED);
            return loadSecretKey(KeyType.ADAL_USER_DEFINED_KEY);
        }

Adam.

Reply:
Yeah but it has to be set on `AuthenticationSettings.INSTANCE` right? ADAL has a public method to do this...but I don't see one in MSAL

Reply:
This is ADAL: https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/cfdb40c5d0fb28e4825df030e909bc493aaaffe9/adal/src/main/java/com/microsoft/aad/adal/AuthenticationSettings.java#L82

Is there something similar in MSAL?

Reply:
PublicClientApplicationConfiguration.java:167

Reply:
Cool, Thanks!